### PR TITLE
refinement of flag parsing (cloud server)

### PR DESCRIPTION
**Pull request by mirshahriar**
## Sample

```go
type InstallationGetFlags struct {
	server           string
	lockInstallation bool
}

func (flags *InstallationGetFlags) AddFlags(cmd *cobra.Command) {
	cmd.Flags().StringVar(&flags.server, "server", "http://localhost:8075", "")
	cmd.Flags().BoolVar(&flags.lockInstallation, "lock-installation", true, "")
}
```

#### Ticket Link

[Jira Ticket](https://mattermost.atlassian.net/browse/MM-48628)

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by fmartingr — APPROVED** [review]
Nice work! I like it way more this way, with everything on the same place :D

If I had to complain about something here is that we do not need many assignments that come from the older flags usage, for example we don't need `slowPoll := sf.slowPoll` anymore since we can just use `sf.slowPoll`, but this seems good to go :D

**Comment by mirshahriar** [comment]
> Nice work! I like it way more this way, with everything on the same place :D
> 
> If I had to complain about something here is that we do not need many assignments that come from the older flags usage, for example we don't need `slowPoll := sf.slowPoll` anymore since we can just use `sf.slowPoll`, but this seems good to go :D

You are right. I will update them all. Thank you for highlighting.

File: cmd/cloud/server_flag.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"time"
+
+	toolsAWS "github.com/mattermost/mattermost-cloud/internal/tools/aws"
+	"github.com/spf13/cobra"
+)
+
+type supervisorOption struct {
```
Comment (gabrieljackson) [inline cmd/cloud/server_flag.go:10]: nit: `supervisorOptions` is probably a bit better along with the other options, but not a big deal. 0/5

**Review by gabrieljackson — APPROVED** [review]
I like this quite a bit! Nice work. I didn't go through every flag string to make sure that nothing was changed so I will trust that your copy-paste skills did the trick. 😄

**Pull request merged**